.buildkite/scripts/common.sh

```diff
@@ -94,9 +100,50 @@ upload_packages_to_gcp_bucket() {
     done
 }
 
+get_bucket_uri() {
+    local type=${1}
+    local baseUri="gs://${JOB_GCS_BUCKET}/jobs/buildkite"
+    if [[ ${type} == "snapshot" ]]; then
+        local folder="commits"
+    else
+        local folder="${type}"
+    fi
+    bucketUri="${baseUri}/${folder}/${BUILDKITE_COMMIT}"
+}
+
+upload_mbp_packages_to_gcp_bucket() {
+    local pattern=${1}
+    local type=${2}
+    get_bucket_uri "${type}"
+    gsutil -m -q cp -a public-read -r ${pattern} ${bucketUri}
+}
+
+download_mbp_packages_from_gcp_bucket() {
+    local pattern=${1}
+    local type=${2}
+    mkdir -p ${WORKSPACE}/${pattern}
+    get_bucket_uri "${type}"
+    gsutil -m -q cp -r ${bucketUri} ${WORKSPACE}/${pattern}
+}
+
+with_mage() {
+    local install_packages=(
+            "github.com/magefile/mage"
+            "github.com/elastic/go-licenser"
+            "golang.org/x/tools/cmd/goimports"
+            "github.com/jstemmer/go-junit-report"
+            "gotest.tools/gotestsum"
+    )
+    create_workspace
+    for pkg in "${install_packages[@]}"; do
+    go install "${pkg}@latest"
+    done
+}
+
 cleanup() {
-    echo "Deleting temporal files..."
-    cd ${WORKSPACE}
-    rm -rf ${TMP_FOLDER_TEMPLATE_BASE}.*
+    echo "Deleting temporary files..."
+    if [[ ! -d "${WORKSPACE}/${TMP_FOLDER_TEMPLATE_BASE}.*" ]]; then
+        rm -rf ${WORKSPACE}/${TMP_FOLDER_TEMPLATE_BASE}.*
+    fi
```

Comment:
I've tried these commands, and it loks like `-d` with a glob does not work.
```
 $ mkdir -p test.12345
 
 $ if [[ -d "test.*" ]] ; then echo "found"; else echo "not found"; fi
not found
```

As `rm -rf`  would not fail in case that template folder (with glob pattern) does not exist, I would suggest to just keep the `rm -rf` without the `if`

```suggestion
    rm -rf ${WORKSPACE}/${TMP_FOLDER_TEMPLATE_BASE}.*
```